### PR TITLE
@W-12316798: [SDK] [Documentation] Add "Common Errors" section to the SDK Docs

### PR DESCRIPTION
**Pull request by saugatt**
- Add docs page for common errors
Screenshot 2023-04-18 at 10 30 03 AM

### CODE REVIEW

File: docs/common-errors.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+If you get 
+  - `Running the plugin script failed`
+  - `Running the plugin script component exceeded the timeout: %1ms`
+  Check that you've correctly implemented `connectionBuilder.js`, `connectionProperties.js` and `required attributes` are correct. Please follow [this](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/postgres_jdbc) example for the Postgres JDBC connector as the starting point for all the files necessary to successfully build a connector. 
+
+
+## Connector Doesn't show in Connector List
+- If your connector does not show up. Check the location of the taco. Follow the section  **Use a connector built with Tableau Connector SDK**
+  in [this doc]([https://help.tableau.com/current/pro/desktop/en-us/examples_connector_sdk.htm)
+- Check that the the taco is signed with a valid certificate. Follow **Sign your packaged connector with jarsigner** section of
```
Comment (pvanderknyff) [inline docs/common-errors.md:25]: Remove this line, the signature being invalid will not stop a connector from appearing on the list

File: docs/common-errors.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+## Connector Doesn't show in Connector List
+- If your connector does not show up. Check the location of the taco. Follow the section  **Use a connector built with Tableau Connector SDK**
+  in [this doc]([https://help.tableau.com/current/pro/desktop/en-us/examples_connector_sdk.htm)
+- Check that the the taco is signed with a valid certificate. Follow **Sign your packaged connector with jarsigner** section of 
+ [Package and Sign Your Connector for Distribution]({{ site.baseurl }}/docs/package-sign). 
+- Check the Tableau logs for any failures for your connector in `/Users/<username>/Documents/My Tableau Repository/Logs/log.txt`
+
+
+## Signature Verification Failed
+If you are getting `Package signature verification failed during connection creation.` in the logs for your connector, you can still use your connector by disabling the signature verification during development by using:
```
Comment (pvanderknyff) [inline docs/common-errors.md:31]: A couple things to point out:
- This error will show up when you click "Sign in" on the connector dialog
- You can see the signature error in the logs, or outside of Tableau by running `jarsigner --verify --strict --verbose --certs [path_to_taco]`

File: docs/common-errors.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Users can search for relevant logs by searching the logs `connector-plugin`, `connector-plugin-warning`, and `connector-plugin-error`. 
+Search for the relevant logs in `/Users/<username>/Documents/My Tableau Repository/Logs/log.txt` for your current instance. 
+[Tableau Log Viewer](https://github.com/tableau/tableau-log-viewer) is a good tool to view and filter relevant logs when developing your connector. 
+
+
+##  Plugin Script Faliure
+If you get 
+  - `Running the plugin script failed`
+  - `Running the plugin script component exceeded the timeout: %1ms`
+  Check that you've correctly implemented `connectionBuilder.js`, `connectionProperties.js` and `required attributes` are correct. Please follow [this](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/postgres_jdbc) example for the Postgres JDBC connector as the starting point for all the files necessary to successfully build a connector.
```
Comment (pvanderknyff) [inline docs/common-errors.md:19]: `required attributes` will not trigger this error, these will only trigger on javascript errors. Also mention that they can see the exact javascript error by searching for `QJSEngine`, which will be nearby to a `connector-plugin-error` log line. Should be easy to fix once they have the exact javascript error.

You should also link to the API reference page (https://tableau.github.io/connector-plugin-sdk/docs/api-reference) and point out the call signatures of the scripts.

File: docs/common-errors.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+---
+title: Common Errors
+---
+**In this section**
+
+* TOC
+{:toc}
+
+## Keywords for Relevant Log Lines.
+Users can search for relevant logs by searching the logs `connector-plugin`, `connector-plugin-warning`, and `connector-plugin-error`.
```
Comment (pvanderknyff) [inline docs/common-errors.md:10]:
```suggestion
You can search for relevant logs by searching the logs `connector-plugin`, `connector-plugin-warning`, and `connector-plugin-error`, or by searching directly for your connector class. 
```

File: docs/common-errors.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  - `Running the plugin script component exceeded the timeout: %1ms`
+  Check that you've correctly implemented `connectionBuilder.js`, `connectionProperties.js` and `required attributes` are correct. Please follow [this](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/postgres_jdbc) example for the Postgres JDBC connector as the starting point for all the files necessary to successfully build a connector. 
+
+
+## Connector Doesn't show in Connector List
+- If your connector does not show up. Check the location of the taco. Follow the section  **Use a connector built with Tableau Connector SDK**
+  in [this doc]([https://help.tableau.com/current/pro/desktop/en-us/examples_connector_sdk.htm)
+- Check that the the taco is signed with a valid certificate. Follow **Sign your packaged connector with jarsigner** section of 
+ [Package and Sign Your Connector for Distribution]({{ site.baseurl }}/docs/package-sign). 
+- Check the Tableau logs for any failures for your connector in `/Users/<username>/Documents/My Tableau Repository/Logs/log.txt`
```
Comment (pvanderknyff) [inline docs/common-errors.md:27]: This should be the first thing they check, we will log the exact reason the taco was not loaded. We log all the file paths for the tacos we've attempted to load under the keyword `connector-plugin`, and they can search for their connector class or their taco's file name to jump to it directly. If they don't see any log lines for their connector, that is a sign they did not place it in the correct place.

File: docs/common-errors.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+
+##  Plugin Script Faliure
+If you get 
+  - `Running the plugin script failed`
+  - `Running the plugin script component exceeded the timeout: %1ms`
+  Check that you've correctly implemented `connectionBuilder.js`, `connectionProperties.js` and `required attributes` are correct. Please follow [this](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/postgres_jdbc) example for the Postgres JDBC connector as the starting point for all the files necessary to successfully build a connector. 
+
+
+## Connector Doesn't show in Connector List
```
Comment (pvanderknyff) [inline docs/common-errors.md:22]: Nitpick, but I'd put this above the Plugin script Failure one, guessing more people will be looking for this.

**Review by pvanderknyff — CHANGES_REQUESTED** [review]

File: docs/common-errors.md (hunk 3 above)
Comment (saugatt) [reply docs/common-errors.md]: Removed the `required attributes` part.  Added API reference page.

File: docs/common-errors.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+**In this section**
+
+* TOC
+{:toc}
+
+## Keywords for Relevant Log Lines.
+You can search for relevant logs by searching the logs `connector-plugin`, `connector-plugin-warning`, and `connector-plugin-error`, or by searching directly for your connector class.
+[Tableau Log Viewer](https://github.com/tableau/tableau-log-viewer){:target="_blank"} is a good tool to view and filter relevant logs when developing your connector. 
+
+## Connector Doesn't show in Connector List
```
Comment (pvanderknyff) [inline docs/common-errors.md:13]:
```suggestion
## Connector Doesn't Show in Connector List
```

File: docs/common-errors.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+- Check the location of the taco. Follow the section  **Use a connector built with Tableau Connector SDK**
+  in [this doc](https://help.tableau.com/current/pro/desktop/en-us/examples_connector_sdk.htm#use-a-connector-built-with-tableau-connector-sdk){:target="_blank"}
+
+##  Plugin Script Faliure
+If you get 
+  - `Running the plugin script failed`
+  - `Running the plugin script component exceeded the timeout: %1ms`
+  Check that you've correctly implemented `connectionBuilder.js` and `connectionProperties.js`  are correct. Please follow [this](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/postgres_jdbc){:target="_blank"} example for the Postgres JDBC connector as the starting point for all the files necessary to successfully build a connector. You can see the exact javascript error by searching for `QJSEngine`, which will be nearby to a `connector-plugin-error` log line.
```
Comment (pvanderknyff) [inline docs/common-errors.md:23]:
```suggestion
  Check that you've correctly implemented `connectionBuilder.js` and `connectionProperties.js`. Please follow [this](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/postgres_jdbc){:target="_blank"} example for the Postgres JDBC connector as the starting point for all the files necessary to successfully build a connector. You can see the exact javascript error by searching for `QJSEngine`, which will be nearby to a `connector-plugin-error` log line.
```

File: docs/common-errors.md (hunk 7 above)
Comment (saugatt) [reply docs/common-errors.md]: Updated this file.

File: docs/common-errors.md (hunk 8 above)
Comment (saugatt) [reply docs/common-errors.md]: Removed `are correct`

**Pull request merged**